### PR TITLE
Update/ add useAutoGeneratedChapter hook

### LINKED ISSUE #5488: Refactor sharable hooks in `edit`
There are a handful of behaviors in `[edit](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx)` that will be shared between the web and native versions. 
In some cases these behaviors include network requests which will require different handlers for the given platform.

### PR DESCRIPTION
**Pull request by jhnstn**
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/5488
Depends on https://github.com/Automattic/jetpack/pull/29024

## Proposed changes:

* Refactor `edit.tsx` to use a custom hook to determine if a Video chapter was auto generated.
* Add a lib file to fetch video chapter data 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

TBD

### CODE REVIEW

File: projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx [hunk 1]
```diff
@@ -144,52 +144,8 @@ export default function VideoPressEdit( {
 	const token = useMediaToken( { guid, isPrivate, tokenType: 'playback' } );
 
 	// Store if the chapters file is auto-generated in a local state.
```
Comment (fluiddot) [inline projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx:146]: I'm wondering if this comment should be along with the following line in the hooks `useAutoGeneratedChapter`, WDYT?
```
const [ isAutoGeneratedChapter, setIsAutoGeneratedChapter ] = useState( true );
```

File: projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx [hunk 2]
```diff
@@ -21,6 +21,7 @@ import debugFactory from 'debug';
  */
 import { isStandaloneActive, isVideoPressActive } from '../../../lib/connection';
 import { buildVideoPressURL, getVideoPressUrl } from '../../../lib/url';
+import useAutoGeneratedChapter from '../../hooks/use-auto-generated-chpater';
```
Comment (fluiddot) [inline projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx:24]: Typo:
```suggestion
import useAutoGeneratedChapter from '../../hooks/use-auto-generated-chapter';
```

File: projects/packages/videopress/src/client/lib/get-chapter/index.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+const getChapter = async ( {
+	guid,
+	token,
+	isPrivate,
+	chapter,
+}: GetChapterArgs ): Promise< string | null > => {
+	if ( ! chapter?.src ) {
+		return;
+	}
+	if ( isPrivate && ! token ) {
```
Comment (fluiddot) [inline projects/packages/videopress/src/client/lib/get-chapter/index.ts:21]: In the previous code this line had the following comment, should we add it back?
```
// Wait for the token when the video is private.
```

File: projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx [hunk 4]
```diff
@@ -144,52 +144,8 @@ export default function VideoPressEdit( {
...
-						setIsAutoGeneratedChapter( isAutoGeneratedChaptersFile );
-					} );
-				} )
-				.catch( function ( error ) {
-					// eslint-disable-next-line no-console
-					console.error( error );
-				} );
-		} catch ( error ) {
-			throw new Error( error?.message ?? error );
-		}
```
Comment (fluiddot) [inline projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx:191]: I'm wondering why we had to catch the `fetch` call using `try-catch` if we already cover the failure of the promise 🤔 .

File: projects/packages/videopress/src/client/lib/get-chapter/index.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+		return;
+	}
+
+	const queryString = token
+		? `?${ new URLSearchParams( { metadata_token: token } ).toString() }`
+		: '';
+
+	const chapterUrl = `https://videos.files.wordpress.com/${ guid }/${ chapter.src }${ queryString }`;
```
Comment (fluiddot) [inline projects/packages/videopress/src/client/lib/get-chapter/index.ts:29]: The type of `guid` argument is marked as optional ([reference](https://github.com/Automattic/jetpack/blob/8dbe986d3d5e2c3712e52d2975ce86e99a2a0a80/projects/packages/videopress/src/client/lib/get-chapter/types.ts#L8)) and I understand that is required to compose the request URL, hence I think we would need either to set it as required or check the value before making the request.

File: projects/packages/videopress/src/client/lib/get-chapter/index.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
+/**
+ * internal dependencies
```
Comment (fluiddot) [inline projects/packages/videopress/src/client/lib/get-chapter/index.ts:2]: Typo:
```suggestion
 * Internal dependencies
```

File: projects/packages/videopress/src/client/block-editor/hooks/use-auto-generated-chpater.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import debugFactory from 'debug';
+/**
+ * WordPress dependencies
+ */
+import getChapter from '../../lib/get-chapter';
+import { VideoChapter } from '../../lib/get-chapter/types';
+import { VideoGUID } from '../blocks/video/types';
+/**
+ * Internal dependencies
+ */
```
Comment (fluiddot) [inline projects/packages/videopress/src/client/block-editor/hooks/use-auto-generated-chpater.ts:14]: Seems we don't have internal dependencies imported in this file so we could remove this comment block.

File: projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx [hunk 8]
```diff
@@ -144,52 +144,8 @@ export default function VideoPressEdit( {
 	const token = useMediaToken( { guid, isPrivate, tokenType: 'playback' } );
 
 	// Store if the chapters file is auto-generated in a local state.
-	const [ isAutoGeneratedChapter, setIsAutoGeneratedChapter ] = useState( true );
-
-	// Detect if the chapter file is auto-generated.
```
Comment (fluiddot) [inline projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx:149]: This comment is missing in the changes, could we incorporate it along with its associated line?

**Review by fluiddot — CHANGES_REQUESTED** [review]

File: projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx [hunk 9]
```diff
@@ -144,52 +144,8 @@ export default function VideoPressEdit( {
...
-				} )
-				.catch( function ( error ) {
-					// eslint-disable-next-line no-console
-					console.error( error );
-				} );
-		} catch ( error ) {
-			throw new Error( error?.message ?? error );
-		}
-	}, [ chapter, guid, token, isPrivate ] );
+	const chapter = tracks?.find( ( track: TrackProps ) => track.kind === 'chapters' )?.[ 0 ];
```
Comment (fluiddot) [inline projects/packages/videopress/src/client/block-editor/blocks/video/edit.tsx:147]: I tested the auto-generated chapters and noticed that they weren't working. Seems the cause is in this function as we changed the function to look for chapters from `filter` to `find`. However, we are still trying to get the first item which results in `undefined` always.

Hence, we either keep the usage of `filter`:
```suggestion
	const chapter = tracks?.filter( ( track: TrackProps ) => track.kind === 'chapters' )?.[ 0 ];
```

Or use `find`:
```suggestion
	const chapter = tracks?.find( ( track: TrackProps ) => track.kind === 'chapters' );
```

**Review by fluiddot — CHANGES_REQUESTED** [review]

**Comment by jhnstn** [comment]
https://github.com/Automattic/jetpack/pull/28945 removes the need for this new hook

**Pull request closed without merging**